src/ballet/block/fd_block.h

```diff
@@ -0,0 +1,176 @@
+#ifndef HEADER_fd_src_ballet_block_fd_block_h
+#define HEADER_fd_src_ballet_block_fd_block_h
+
+/* Blocks are the logical representation of Solana block data.
+
+   They consist of 64 entries, each containing a vector of txs. */
+
+#include "../fd_ballet_base.h"
+#include "../sha256/fd_sha256.h"
+
+#include "../txn/fd_txn.h"
+
+#define FD_ENTRY_ALIGN (64UL)
+
+#define FD_ENTRY_MAGIC (0x176aa423e0372d6aUL) /* random */
+
+/* fd_entry induces a PoH tick and contains a set of transactions.
+
+   ### PoH Ticks
+
+   Because each slot has 64 ticks, each block has 64 entries.
+
+   The PoH tick is executed by first appending `hash_cnt-1` hashes to
+   the PoH state, and then mixing in the `entry hash`.
+
+   ### Entry Hash
+
+   The entry hash is the 32-byte root of the binary Merkle tree given
+   the first signature of each transaction as an input. */
+
+/* fd_entry_hdr: Fixed size header preceding entry payload. */
+struct __attribute__((packed)) fd_entry_hdr {
+  /* Number of PoH hashes between this entry and last entry */
+  /* 0x00 */ ulong hash_cnt;
+
+  /* PoH state of last entry */
+  /* 0x08 */ uchar hash[ FD_SHA256_HASH_SZ ];
+
+  /* Number of transactions in this entry */
+  /* 0x28 */ ulong txn_cnt;
+};
+typedef struct fd_entry_hdr fd_entry_hdr_t;
+
+FD_STATIC_ASSERT( sizeof(fd_entry_hdr_t)==48UL, alignment );
+
+/* `fd_txn_o` (read `fd_txn` owned) is a buffer that fits any `fd_txn_t`. */
+struct fd_txn_o {
+  /* Buffer containing `fd_txn_t`. */
+  uchar txn_desc_raw[ FD_TXN_MAX_SZ ];
+};
+typedef struct fd_txn_o fd_txn_o_t;
+
+/* `fd_rawtxn_b` references a serialized txn backing an `fd_txn_t`. */
+struct fd_rawtxn_b {
+  /* Pointer to txn in local wksp */
+  void * raw; /* TODO: Make this a gaddr instead of laddr */
+
+  /* Size of txn */
+  ushort txn_sz;
+};
+typedef struct fd_rawtxn_b fd_rawtxn_b_t;
+
+/* fd_entry: Buffer storing a deserialized entry. */
+struct __attribute__((aligned(FD_ENTRY_ALIGN))) fd_entry {
+  /* This point is 64-byte aligned */
```

Comment:
Is there a value in making fd_entry less ambiguous?   s/fd_entry/fd_block_entry/